pkg/synopsysctl/cmd_create.go

```
@@ -526,11 +535,165 @@ var createOpsSightNativeCmd = &cobra.Command{
+var createPolarisNativeCmd = &cobra.Command{
+	Use:           "native NAME",
+	Example:       "synopsysctl create polaris native <name>",
+	Short:         "Print the Kubernetes resources for creating a Polaris instance",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Check the Number of Arguments
+		if len(args) != 0 {
+			cmd.Help()
+			return fmt.Errorf("this command takes 0 arguments")
+		}
+		return nil
+	},
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		err := createPolarisCobraHelper.SetPredefinedCRSpec("")
+		if err != nil {
+			cmd.Help()
+			return err
+		}
+		cobra.MarkFlagRequired(cmd.Flags(), "version")
+		cobra.MarkFlagRequired(cmd.Flags(), "environment-dns")
+		cobra.MarkFlagRequired(cmd.Flags(), "environment-name")
+		cobra.MarkFlagRequired(cmd.Flags(), "postgres-username")
+		cobra.MarkFlagRequired(cmd.Flags(), "postgres-password")
+
+		cobra.MarkFlagRequired(cmd.Flags(), "smtp-host")
+		cobra.MarkFlagRequired(cmd.Flags(), "smtp-port")
+		cobra.MarkFlagRequired(cmd.Flags(), "smtp-username")
+		cobra.MarkFlagRequired(cmd.Flags(), "smtp-password")
+
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		polarisObj, err := updatePolarisSpecWithFlags(cmd, namespace)
+		if err != nil {
+			return err
+		}
+
+		var objectArr []interface{}
+		for _, v := range polaris.GetComponents(baseUrl, *polarisObj) {
+			objectArr = append(objectArr, v)
+		}
+
+		PrintComponents(objectArr, nativeFormat)
+
+		return nil
+	},
+}
+
+func updatePolarisSpecWithFlags(cmd *cobra.Command, namespace string) (*polaris.Polaris, error) {
+	// Update Spec with user's flags
+	log.Debugf("updating spec with user's flags")
+	polarisInterface, err := createPolarisCobraHelper.GenerateCRSpecFromFlags(cmd.Flags())
+	if err != nil {
+		return nil, err
+	}
+
+	polarisSpec, ok := polarisInterface.(polaris.Polaris)
+	if !ok {
+		panic("Couldn't cast polarisInterface to polarisSpec")
```

Why panic here? Shouldn't we just return an error message?

---

I will improve error handling and messages in a different PR